crates/sparrow-instructions/src/evaluators/aggregation/token/collect_struct_token.rs

```diff
@@ -28,19 +41,45 @@ impl StateToken for CollectStructToken {
 
 impl CollectStructToken {
     pub fn new(state: ArrayRef) -> Self {
-        Self { state }
+        let fr = Arc::new(Field::new(
+            "item",
+            DataType::Timestamp(TimeUnit::Nanosecond, None),
+            true,
+        ));
+        let times_type = DataType::List(fr);
+        Self {
+            state,
+            times: new_empty_array(&times_type),
+        }
+    }
```

Comment:
Probably should split this up? But didn't want to have multiple `CollectStructEvaluators`. Unsure what the best separation is at the moment, so just using this as a combined token, and the non-windowed and since-windowed evaluators ignore the `times` array.

Reply:
Minor performance/memory hit in that we have to resize the `times` array, but it's all `nulls`.